core/src/main/java/io/questdb/cairo/pool/WriterPool.java

```diff
@@ -197,12 +196,17 @@ public boolean lock(CharSequence tableName) {
         // try to change owner
         if ((Unsafe.cas(e, ENTRY_OWNER, UNALLOCATED, thread) /*|| Unsafe.cas(e, ENTRY_OWNER, thread, thread)*/)) {
             closeWriter(thread, e, PoolListener.EV_LOCK_CLOSE, PoolConstants.CR_NAME_LOCK);
-            return lockAndNotify(thread, e, tableName);
+            if (lockAndNotify(thread, e, tableName, lockReason)) {
+                return null;
+            }
+            assert null != e.ownershipReason;
+            return e.ownershipReason;
```

Comment:
I'm paranoiac and  despite the asserts would change to
```
return e.ownershipReason != null ? e.ownershipReason : "UNKNOWN";
```

Reply:
Well we have very high test coverage around this area. What might be a good idea is to through a null pointer exception if someone tries to set a null reason